src/NHibernate/Criterion/ISupportEntityJoinQueryOver.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Linq.Expressions;
+using NHibernate.SqlCommand;
+
+namespace NHibernate.Criterion
+{
+	//TODO: Make interface more flexible for changes (maybe it should take only 2 params alias + EntityJoinConfig)
+	public interface ISupportEntityJoinQueryOver
```

Comment:
I would add a " 6.0 TODO: merge with IQueryOver". Is there a reason for not aiming at this? The same apply to below interface, but of course with `IQueryOver<TRoot>`.